packages/components/button/button.js

```diff
@@ -1,10 +1,21 @@
+const KEY_SPACE = 32
+const DEBOUNCE_TIMEOUT_IN_SECONDS = 1
+
 class Button {
   constructor($module) {
-    this.KEY_SPACE = 32
-    this.DEBOUNCE_TIMEOUT_IN_SECONDS = 1
+    if (!$module) {
+      return this
```

Comment:
Not a blocker, but this behaviour of returning a partially instantiated object if called with invalid arguments or dom structure seems a bit strange to me. Is it something you're planning to revisit?

I would expect it to throw an error so it can be handled if necessary, similar to the type checking here https://github.com/alphagov/govuk-frontend/blob/main/packages/govuk-frontend/src/govuk/component.mjs#L60